apiml-security-common/src/main/java/org/zowe/apiml/security/common/verify/CertificateValidator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.security.common.verify;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.zowe.apiml.message.log.ApimlLogger;
+import org.zowe.apiml.product.logging.annotations.InjectApimlLogger;
+
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.List;
+
+/**
+ * Service to verify if given certificate chain can be trusted.
+ */
+@Service
+@Slf4j
+public class CertificateValidator {
+
+    final TrustedCertificatesProvider trustedCertificatesProvider;
+
+    @InjectApimlLogger
+    private final ApimlLogger apimlLog = ApimlLogger.empty();
+
+    @Getter
+    @Value("${apiml.security.x509.acceptForwardedCert:false}")
+    private boolean forwardingEnabled;
+
+    @Value("${apiml.security.x509.certificatesUrl:}")
+    private String proxyCertificatesEndpoint;
```

Comment:
this value can be moved to TrustedCertificatesProvider since it is not used in this class.

Reply:
It might happen that TrustedCertificatesProvider.getTrustedCerts() can be called against different URL (different cloud gateways). And the url as method parameter also serves as a key for the cached result of this method.